Concepts overhaul: consolidate basic concepts into 2 page overview, ideas and theory (#1457)
ipfs/ipfs-docs: docs/concepts/what-is-ipfs.md

---

> title: What IPFS is and isn't

Reviewer: ```suggestion
title: What is IPFS?
```

I think "What is IPFS?" is clearer and also works better for SEO

:::callout
This guide is part 1 of a 3-part introduction to [the basic concepts of IPFS](../concepts/README.md#learn-the-basics). The second part, [**IPFS and the problems it solves**](../concepts/ipfs-solves.md), covers the problems with the internet and current protocols like HTTP that IPFS solves.
:::

In this conceptual guide, you'll learn what IPFS is and isn't.

## Defining IPFS

The term _IPFS_ can refer to multiple concepts:

> - Software that implements the IPFS protocol specification (like <VueCustomTooltip label="The first implementation of IPFS, written in Go." underlined multiline>Kubo</VueCustomTooltip>) 

Reviewer: ```suggestion
- Software that implements the IPFS protocol specifications (like <VueCustomTooltip label="The first implementation of IPFS, written in Go." underlined multiline>Kubo</VueCustomTooltip>) 
```

Can you get the idea across that there are multiple specs and not all implementations implement all of them.

---
title: What is IPFS?
description:  Learn about the problems that IPFS addresses, what IPFS is and isn't, and how the different subsystems that IPFS is comprised of work.
---

# What is IPFS?

IPFS is a modular suite of protocols and standards for organizing and moving data, designed from the ground up with the principles of <VueCustomTooltip label="A way to address data by its hash rather than its location (IPs)." underlined>content addressing</VueCustomTooltip> and peer-to-peer networking. 

:::callout

> This guide is part 1 of a 2-part introduction to the basic concepts of IPFS. For the second part, which covers how the different subsystems that IPFS is comprised of work and how they work, see [How IPFS works](../concepts//how-ipfs-works.md).

Reviewer: ```suggestion
This guide is part 1 of a 2-part introduction to the basic concepts of IPFS. See the second part: [**How IPFS Works**](../concepts//how-ipfs-works.md), which covers how the different subsystems that IPFS is comprised and how they work.
```

:::

In this conceptual guide, you'll learn:

- [What IPFS is and isn't](#definining-ipfs)
- [Problems that IPFS addresses](#problems-that-ipfs-addresses)

## Definining IPFS

The term _IPFS_ can refer to multiple concepts:

- Software that implements the IPFS protocol specification (like <VueCustomTooltip label="The first implementation of IPFS, written in Go." underlined>Kubo</VueCustomTooltip>) 

> - A decentralized network comprised of <VueCustomTooltip label="Computers participating in an IPFS network by running an IPFS implementation. Also referred to as peers" underlined>IPFS nodes</VueCustomTooltip>.

Reviewer: WDYT about this to emphasise the idea of participation. 

```suggestion
- A decentralized network comprised of <VueCustomTooltip label="Computers participating in an IPFS network by running an IPFS implementation. Also referred to as peers" underlined>IPFS nodes</VueCustomTooltip> that anyone can join.
```

- A modular suite of protocols and standards for organizing and moving content-addressed data.

### What IPFS _isn't_

While IPFS shares similarities with, and is often used in architectures with the systems described below, IPFS is _not_:

- _A <VueCustomTooltip label="A list of records, known as blocks, linked using cryptography. Each block contains a cryptographic hash of the previous block, a timestamp, and transaction data." underlined multiline is-medium>blockchain</VueCustomTooltip>_: While IPFS and blockchains do share some concepts and technologies (the use of <VueCustomTooltip label="A function that takes some arbitrary input (content) and returns a fixed-length value. The exact same input data will always generate the same hash as output. There are numerous hash algorithms." underlined multiline is-medium>cryptographic hashes</VueCustomTooltip>, decentralization, verifiability of data, etc.), and IPFS is is used for [verifiable off-chain storage](#verifiable-off-chain-storage), IPFS is not a blockchain.

> - A _storage system, network or provider_: Rather, IPFS  provides a modular suite of protocols and standards for organizing and moving content-addressed data, which many storage solutions and other applications use as a fundamental part of their architecture.

Reviewer: ```suggestion
- A _storage provider_: Rather, IPFS  provides a modular suite of protocols and standards for organizing and moving content-addressed data, which many storage solutions and other applications use as a fundamental part of their architecture.
```

We want to tread this carefully. In a sense, it is a network, so I don't want to confuse reader too much.

- [Centralization](#centralization)
- [Performance](#performance)
- [Scalability](#scalablity)
- [Link rot](#link-rot)
- [Security](#security)
- [Data sovereignty and ownership](#data-sovereignty)
- [Verifiable off-chain storage](#verifiable-off-chain-storage)
- [Local-first software](#local-first-software)
- [Vendor lock-in](#vendor-lock-in)

### Centralization

IPFS is an open-source, decentralized system that eliminates the need for a central authority, making it more resilient and censorship-resistant than traditional centralized systems. No single entity or person controls, manages or owns IPFS; rather, it is a community-maintained project with multiple implementations of the protocol, multiple tools and apps leveraging that protocol, and multiple users and organizations contributing to it's design and development.

### Performance

> IPFS provides faster access to data by storing it at multiple locations, and allowing users to access it from the nearest location using content addressing instead of <VueCustomTooltip label="Data identified and linked to by it's location. An example is HTTP." underlined>location-based addressing</VueCustomTooltip>. In other words, because data can be addressed based on it's contents, a node on the network can fetch that data from _any_ other node in the netork that has the data; thus, performance issues like latency are reduced. 

Reviewer: **IPFS itself doesn't store in multiple locations.**

The way I think about it is that "IPFS enshrines the freedom to replicate data on the network" and that "IPFS enables data to be retrieved from multiple locations"

> ### Scalability

Reviewer: ```suggestion
### Resilience 
```

Also adding this screenshot of copy from the new IPFS website which we should integrate into this page
<img width="669" alt="Screenshot 2023-02-03 at 2 38 11 PM" src="https://user-images.githubusercontent.com/1992255/216617080-66dae58b-5965-4ba7-b4d0-cddda2b1f3c0.png">

IPFS has no single point of failure, and users do not need to trust each other. In other words, the failure of a single or even multiple nodes in the network does not affect the functioning of the entire network, and 

> ### Link rot

Reviewer: This one feels a little weak. This one is an important point but could be more refined.

IPFS eliminates the problem of broken links by allowing data to be addressed by its content, rather than by its location.

> ### Security

Reviewer: For consistency with the website, I would stick to **verifiable**.

IPFS uses <VueCustomTooltip label="A function that takes some arbitrary input (content) and returns a fixed-length value. The exact same input data will always generate the same hash as output. There are numerous hash algorithms." underlined multiline is-medium>cryptographic hashes</VueCustomTooltip> to ensure the authenticity and integrity of files, making it difficult for malicious actors to tamper with or delete files.

### Data sovereignty 

IPFS protects <VueCustomTooltip label="The idea that individuals or organizations have control over their own data and the ability to determine who can access and use it." underlined multiline is-medium>data sovereignty</VueCustomTooltip> by enabling users to store and access data directly on a decentralized network of nodes, rather than centralized, third-party servers. This eliminates the need for intermediaries to control and manage data, giving users full control and ownership over their data.

> ### Verifiable off-chain storage

Reviewer: If we rename "security" above to verifiability, this can be simply off-chain storage because verifiability is implied.

> IPFS enables verifiable <VueCustomTooltip label="Storage outside of a blockchain for data processed by the blockchain. Used to store large amounts of data that would be inefficient to store directly on a blockchain, improving scalability and efficiency." underlined multiline is-medium>off-chain storage</VueCustomTooltip> by linking the off-chain data to an IPFS content-addressed data identifier (a <VueCustomTooltip label="An address used to point to data in IPFS, based on the content itself, as opposed to the location." underlined multiline is-medium>Content IDentifier (CID)</VueCustomTooltip>, explained in [How IPFS works](#content-identifier-cid)), which is stored on the IPFS network. This identifier contains a cryptographic hash of the data, allowing anyone to verify the integrity of the data without having to retrieve the data.

Reviewer: ```suggestion
IPFS enables verifiable <VueCustomTooltip label="Storage outside of a blockchain for data processed by the blockchain. Used to store large amounts of data that would be infeasible to store directly on a blockchain, improving scalability and efficiency." underlined multiline is-medium>off-chain storage</VueCustomTooltip> by creating a link between blockchain state and content-addressed published to IPFS. This works by storing a <VueCustomTooltip label="An address used to point to data in IPFS, based on the content itself, as opposed to the location." underlined multiline is-medium>Content IDentifier (CID)</VueCustomTooltip>(explained in [How IPFS works](#content-identifier-cid)) in a smart contract. 
```

I removed:
>The CID contains a cryptographic hash of the data, allowing anyone to verify the integrity of the data without having to retrieve the data.

Because it's not actually true

### Local-first software

IPFS benefits <VueCustomTooltip label="Software in which data is stored and processed locally, and is then synchronized and shared with other devices when a network connection is available. By keeping data local, local-first software reduces dependency on internet connectivity, and emphasizes data sovereignty and privacy." underlined multiline is-medium>local-first software</VueCustomTooltip> by providing a performant, decentralized, peer-to-peer data addressing, routing, and transfer protocol that prioritizes data storage and processing on individual devices. With IPFS, data can be stored, verified and processed locally, and then synchronized and shared with other IPFS nodes when a network connection is available.

> ### Vendor lock-in

Reviewer: In this section, I think we can hone in on another property of IPFS
- content-addressing, which decouples the address from a single location, thereby enabling changing storage locations without changing APIs and how you manage your data. In contrast, if you move from AWS to GCloud, you need to integrate new APIs.
